crates/subspace-farmer/src/utils/farmer_provider_storage.rs

```diff
@@ -107,7 +109,7 @@ where
                 key: piece_index_hash.to_multihash().into(),
                 provider: self.local_peer_id,
                 expires: None,
-                addresses: Vec::new(), // TODO: add address hints
+                addresses: Vec::new(), // Kademlia adds addresses for local providers
```

Comment:
Comment in libp2p makes perfect sense, now I understand what you meant, but you better copy the whole comment from libp2p or re-phrase it, indicating that Kademlia will add it later for locally provided records. The way it is written right now is confusing.

Which in turn doesn't make sense anymore because we have supposedly removed Kademlia's implementation 🤷

Reply:
I copied the comment. We removed Kademlia implementation only for announcements because we had issues with it. `get_providers` operation works well and it will be suboptimal to replace it with `get_closest_peers` + custom requests. Please, let me know if I understood you incorrectly.